modules/renter/renter.go

```diff
@@ -155,36 +144,41 @@ type trackedFile struct {
 
 // A Renter is responsible for tracking all of the files that a user has
 // uploaded to Sia, as well as the locations and health of these files.
+//
+// TODO: Separate the workerPool to have its own mutex. The workerPool doesn't
+// interfere with any of the other fields in the renter, should be fine for it
+// to have a separate mutex, that way operations on the worker pool don't block
+// operations on other parts of the struct. If we're going to do it that way,
+// might make sense to split the worker pool off into it's own struct entirely
+// the same way that we split of the memoryManager entirely.
 type Renter struct {
 	// File management.
 	//
 	// tracking contains a list of files that the user intends to maintain. By
 	// default, files loaded through sharing are not maintained by the user.
 	files    map[string]*file
-	tracking map[string]trackedFile // map from nickname to metadata
+	tracking map[string]trackedFile // Map from nickname to metadata.
 
-	// Work management.
-	//
-	// chunkQueue contains a list of incomplete work that the download loop acts
-	// upon. The chunkQueue is only ever modified by the main download loop
-	// thread, which means it can be accessed and updated without locks.
+	// Download management. The heap has a separate mutex because it is always
+	// accessed in isolation.
+	downloadHeapMu sync.Mutex         // Used to protect the downloadHeap.
+	downloadHeap   *downloadChunkHeap // A heap of priority-sorted chunks to download.
+	newDownloads   chan struct{}      // Used to notify download heap thread that new downlaods are available.
```

Comment:
with a bit of work, `newDownloads` could be moved inside the `downloadChunkHeap` type, and you could give `downloadChunkHeap` a method for signaling new downloads. Then the implementation details around signal handling are hidden.